jobs/otel-collector-windows/templates/config.yml.erb

```
@@ -44,7 +44,7 @@ def check_for_unused_secrets!(secrets, used_secrets)
 end
 
 def check_for_new_and_old_properties!
-  both_provided = !p('config').empty? && (!p('metric_exporters').empty? || !p('trace_exporters').empty?)
+  both_provided = (!p('config').empty? || !p('prom_exporter_config').empty?) && (!p('metric_exporters').empty? || !p('trace_exporters').empty?)
```

We should bind the configuration to the selection of the other exporters. Can you please adapt it?

```suggestion
  both_provided = (!p('config').empty? && (!p('prom_exporter_config').empty?) || !p('metric_exporters').empty? || !p('trace_exporters').empty?)
```

---

Sorry, I am not clear about the ask. The code here is to ensure deprecated config (metric_exporters, trace_exporters) and "config" can't be set at the same time. The new prom_exporter_config will be merged into "config", so the presence of prom_exporter_config or "config" indicates we are using the new config.